spyder/plugins/tests/test_editor_introspection.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+"""Tests for the Editor plugin."""
+
+# Third party imports
+import pytest
+import os
+import os.path as osp
+from flaky import flaky
+
+try:
+    from unittest.mock import Mock
+except ImportError:
+    from mock import Mock  # Python 2
+
+from qtpy.QtWidgets import QWidget, QApplication
+from qtpy.QtCore import Qt
+
+from spyder.utils.introspection.jedi_plugin import JEDI_010
+from spyder.utils.qthelpers import qapplication
+
+# Location of this file
+LOCATION = osp.realpath(osp.join(os.getcwd(), osp.dirname(__file__)))
+
+
+@pytest.fixture
+def setup_editor(qtbot, monkeypatch):
+    """Set up the Editor plugin."""
+    app = qapplication()
+    os.environ['SPY_TEST_USE_INTROSPECTION'] = 'True'
+    monkeypatch.setattr('spyder.dependencies', Mock())
+    from spyder.plugins.editor import Editor
+
+    monkeypatch.setattr('spyder.plugins.editor.add_actions', Mock())
+
+    class MainMock(QWidget):
+        def __getattr__(self, attr):
+            if attr.endswith('actions'):
+                return []
+            else:
+                return Mock()
+
+        def get_spyder_pythonpath(*args):
+            return []
+
+    editor = Editor(MainMock())
+    qtbot.addWidget(editor)
+    editor.show()
+    editor.new(fname="test.py", text="")
+    editor.introspector.set_editor_widget(editor.editorstacks[0])
+
+    yield editor, qtbot
+    # teardown
+    os.environ['SPY_TEST_USE_INTROSPECTION'] = 'False'
+    editor.introspector.plugin_manager.close()
+
+
+@pytest.mark.skipif(True, reason="This test fails too much in the CI :(")
```

Comment:
Let's remove this condition and change the one below to

    not JEDI_010 or os.environ.get('CI', False)

so we can run this test locally.